Stop using the intel driver

dnschneid opened the pull request:
Past me says this:

> Use modesetting; glamor will happen by default if available.
> Compatibility with Intel DRI has deteriorated significantly with Freon.
> Using modesetting will be the most compatible and only has a minor performance hit.

I don't remember what triggered me to make these patches, nor do I recall the result of testing or whether this is actually ready to merge.

I figure creating a PR will help figure things out...

EnderNightLord-ChromeBook commented:
Good Idea Is it possible that this could be a target (as in you can disable and enable)

tblaschke commented:
This fixes the missing characters using Xorg on a Samsung Chromebook Pro. Maybe we should test it on other Skylake Chromebooks and activate it by default?

dnschneid commented on targets/xorg:
```diff
@@ -207,35 +109,12 @@ if [ -n "$removevideodrivers" ]; then
-    # instead of geteuid, to force DRM master setting/dropping.
-    offset="`grep -F -m 1 -boa 'geteuid' "$inteldrv" 2>/dev/null || true`"
-    if [ -n "$offset" ]; then
-        echo -n 'getuid0' | dd seek="${offset%:*}" bs=1 of="$inteldrv" \
```
Remove getuid0 from freon.c?

octopusnz commented:
Sounds like this might help with #2958 ? If so that is good news.

dnschneid commented:
I can make a branch release with this patch if you want to try it.

dnschneid commented:
You can give it a shot by updating your chroot with `sudo CROUTON_BRANCH=nointeldrv crouton -u -n chrootname`

kelchm commented:
Related to #2640.

dnschneid commented:
@kelchm @octopusnz does this branch fix things for either of you?

octopusnz commented:
@dnschneid I think this does resolve #2958 for me, or at least has decreased the frequency.

kelchm commented:
@dnschneid, it did not work for me the last time I tried -- let me give it another go tonight.

kelchm commented:
@dnschneid, I've not been able to get this branch to work. Here's what I'm seeing:

```
sudo CROUTON_BRANCH=nointeldrv sh ~/Downloads/crouton -u -n xenial-gnome
...
Use 'apt autoremove' to remove it.
The following packages will be REMOVED:
  xserver-xorg-video-intel*
The following held packages will be changed:
  xserver-xorg-video-intel
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
E: Held packages were changed and -y was used without --allow-change-held-packages.
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/xenial-gnome...
Sending SIGTERM to processes under /mnt/stateful_partition/crouton/chroots/xenial-gnome...
```

octopusnz commented:
Actually after playing with it for a few weeks - issue still exists (Chrome flickering).
I'm assuming I managed to install from the right branch but based on the below maybe not?

I think for this to fix my issue it needs to stop using the i915 driver for the video card.

$ dpkg -s xserver-xorg-video-intel | grep Status
Status: hold ok installed

$ lshw -c video
WARNING: you should run this program as super-user.
  *-display               
       description: VGA compatible controller
       product: Atom Processor Z36xxx/Z37xxx Series Graphics & Display
       vendor: Intel Corporation
       physical id: 2
       bus info: pci@0000:00:02.0
       version: 0e
       width: 32 bits
       clock: 33MHz
       capabilities: vga_controller bus_master cap_list rom
       configuration: driver=i915 latency=0
       resources: irq:262 memory:d0000000-d03fffff memory:c0000000-cfffffff ioport:1000(size=8)


Possible I did something wrong when trying to install from the branch however.
@dnschneid Is there a logfile for crouton for installs/upgrades that would let me easily check if I managed to install from a branch rather than master?

dnschneid commented:
@octopusnz running `croutonversion` inside the chroot, or `sudo edit-chroot -all` from outside spit out the version strings, which include which branch was installed.

ammaarpatel99 commented:
has anyone switched to the no intel driver branch and has successfully stopped the screen flickering. im using the unity version of trusty on crouton and after the latest update the croot wouldnt load. i had to power wash my pixelbook, leave it for a few days and then reinstall crouton. the croot now loads but the screen and the curser flicker constantly with large section of the screen sometimes going black for a few seconds.

rossjrw commented:
Using this branch I've had major screen tearing issues and poor performance even just when alt+tabbing, with menu navigation sometimes reaching 80% on my cpu. Games on steam that I could previously play at ~60fps I now struggle to reach 10.

On the other hand, without the fix from this branch crouton doesn't work at all.

Any suggestions for getting my performance back?

ammaarpatel99 commented:
I turned on OS verification last week and I have used it since then. Yesterday I went on it and restarted to upgrade flash something out another. Then I enabled developer mode and downloaded crouton from the GitHub page. I created an encrypted trusty unity chroot with touch enabled. Now the chroot works like a dream. I don't know what's changed but something is definitely different. I hope this helps.

pmarkkula commented:
@rossjrw 

You might want to check that you actually have hw accelerated opengl. Modesetting and glamor work also with sw fallback.

rossjrw commented:
@pmarkkula Thanks for your response - do you know how I'd check this?

pmarkkula commented:
@rossjrw 

Try 

`glxinfo | grep -E "Vendor|Device"`

 in chroot. Mine says:

```
    Vendor: Intel Open Source Technology Center (0x8086)
    Device: Mesa DRI Intel(R) Broadwell GT1  (0x1606)
```

rossjrw commented:
@pmarkkula Many thanks for your help. The command you suggested had no output, although I also tried a lowercase version `glxinfo | grep -E "vendor|device"` which had the output

    server glx vendor string: SGI
    client glx vendor string: Mesa Project and SGI
    OpenGL vendor string: VMware, Inc.

Assuming this means I'm SW accelerated, how would I correct this?

pmarkkula commented:
@rossjrw 

You might be missing the appropriate driver package.

```apt-cache policy xserver-xorg-video-intel```

should tell if this is the case.

rossjrw commented:
@pmarkkula the output of that command is the following:

```
xserver-xorg-video-intel:
  Installed: 2:2.99.910-0ubuntu1.7
  Candidate: 2:2.99.910-0ubuntu1.7
  Version table:
 *** 2:2.99.910-0ubuntu1.7 0
        500 http://archive.ubuntu.com/ubuntu/ trusty-updates/main amd64 Packages
        100 /var/lib/dpkg/status
     2:2.99.910-0ubuntu1 0
        500 http://archive.ubuntu.com/ubuntu/ trusty/main amd64 Packages
```

I have no clue what to make of this - could you shed some wisdom?

pmarkkula commented:
@rossjrw 

So the Intel driver is installed. If it would not have been, it would have said something like "Installed: (none)". I am running out of ideas.

rossjrw commented:
@pmarkkula That's fair - thank you for all the help you've given anyway!

gavinfernandes2012 commented on targets/xorg:
```diff
@@ -207,35 +174,12 @@ if [ -n "$removevideodrivers" ]; then
-    rm -f "$xorgconf"
-fi
+# Remove old Intel SNA workaround
+rm -f '/usr/share/X11/xorg.conf.d/20-crouton-intel-sna.conf'
```
Are there any replacements for this workaround? It seems like *bionic* needs this workaround on my system (CBPro) with gnome/kde to fix #3988 / #3933

bline commented:
Has there been any movement on this? I just installed xenial xfce with the latest crouton and have a lot of flickering from chrome, not really usable. This is on skylake c302 asus flip.
Somehow this branch will not install. The command:
`sudo CROUTON_BRANCH=nointeldrv crouton -u`
downloads a github 404 page.

dnschneid commented:
> Has there been any movement on this? I just installed xenial xfce with the latest crouton and have a lot of flickering from chrome, not really usable. This is on skylake c302 asus flip.
> Somehow this branch will not install. The command:
> `sudo CROUTON_BRANCH=nointeldrv crouton -u`
> downloads a github 404 page.

I had removed the release because it was terribly out-of-date.
I just rebased and pushed a new release for the branch.  You're welcome to try it but I have no idea if it'll work at all.

bline commented:
Thanks for the update. I ran the update to this branch mentioned above with this error:
```
The following packages will be REMOVED:
  xserver-xorg-video-intel*
The following held packages will be changed:
  xserver-xorg-video-intel
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
E: Held packages were changed and -y was used without --allow-change-held-packages.
Failed to complete chroot setup.
```
Can I manually run this remove or was there more that needs to happen after that?

bline commented:
I went ahead and did that remove. startxfce4 went to grey screen with mouse and locked my system.

dnschneid commented:
Great success :/

Can you please create a new chroot with it just to be sure it's totally broken?

Unfortunately I don't have the devices or time to get this tested and working at the moment, but if you find some tweaks that make things work I can incorporate it into the branch.

bline commented:
So I ran this from a new chroot with the following command:
`sudo CROUTON_BRANCH=nointeldrv crouton -t xfce,touch,cli-extra,chrome,audio,extension -n xenial2`
It failed with:
```
Compiling /usr/local/lib/croutonfreon.so...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package 'xserver-xorg-input-vmmouse' is not installed, so not removed
Package 'xserver-xorg-video-all' is not installed, so not removed
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 xorg : Depends: xserver-xorg (>= 1:7.7+13ubuntu3.1)
E: Unable to correct problems, you have held broken packages.
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/xenial2...
```

I would be happy to help in anyway you need.

bline commented:
Oh, and again, `sudo startxfce4` gets to grey screen and locks system.

TomTravis commented:
it not clear the intel drive is used on kali bookworm and jammy
many KMS kernel patches and made and the driver maybe updated 
for my fork xwiw has issues with color map and xrandr selection
the xorg log for digital display and the correct xrandr for working xiwi configs
xorg does not start in the crouton install Kail-rolling bookworm and jammy, kali has the best video by far
